### PR TITLE
Added HSTS plugin documentation and structure for adding documentation about other plugins in the future.

### LINKED ISSUE #51: HSTS Plugin documentation
There should be documentation for the HSTS plugin in `/docs/`. There should also be some document listing all plugins that are provided by the framework.

### PR DESCRIPTION
**Pull request by mattiasgrenfeldt**
Fixes #51 

Documentation for the HSTS plugin has been added. Also added a `plugins.md` file which can be used to describe plugins more in the future. `plugins.md` also contains a list of all the plugins we have currently implemented.

I also updated a godoc comment in the `hsts.Plugin` struct.

The links will point to the right place once this is merged.

### CODE REVIEW

File: docs/high-level-design.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ address, following issues (not an exhaustive list):
 - [**CSP (content security policy)**](https://csp.withgoogle.com/docs/index.html) -
   e.g. by automatically adding script nonces to HTML responses, adding relevant
   security headers
-- **Transport Security** - e.g. by enforcing HSTS support
+- **Transport Security** - e.g. by [enforcing HSTS support](https://github.com/google/go-safeweb/blob/master/docs/plugins/hsts.md)
```
Comment (mihalimara22) [inline docs/high-level-design.md:77]: This link is not working for me.

File: docs/plugins.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+_Authors: grenfeldt@google.com_
+
+# Plugins
+
+The framework will support plugins. Plugins can run before and after the request handlers. They provide a flexible way to address security issues by setting non-overridable security headers or interrupting and responding to incoming requests before they reach the handler.
```
Comment (mihalimara22) [inline docs/plugins.md:5]: "will support plugins that run before and after.. "

File: docs/plugins/hsts.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+_Authors: grenfeldt@google.com_
+
+# HSTS Plugin
+
+HSTS<sup>1</sup> (HTTP Strict Transport Security) tells browsers that they should only access your site using HTTPS and not HTTP. This plugin enforces HSTS by redirecting all HTTP traffic to HTTPS and by setting the `Strict-Transport-Security` header on all outgoing HTTPS traffic.
```
Comment (mihalimara22) [inline docs/plugins/hsts.md:5]: "... informs browsers that a website should only be accessed using HTTPS and not HTTP"

File: docs/plugins/hsts.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+To construct the plugin with safe default settings, use: `hsts.NewPlugin()`.
+
+## Options
+
+**Option (Default value)**: Description.
+
+- **MaxAge (`2 years`)**: The amount of time that the browser should remember to only access this site using HTTPS. The default value is 2 years since this is part of the deployment recommendations of https://hstspreload.org/.
+- **IncludeSubDomains (`enabled`)**: When this is enabled, the browser knows that all subdomains should also only be accessed via HTTPS.
+- **Preload (`disabled`)**: Must be enabled for this domain to eligible to be added to the browser HSTS preload list. For more info, see: https://hstspreload.org/. This is disabled by default so that your domain can't be added to the preload list unintentionally.
```
Comment (mihalimara22) [inline docs/plugins/hsts.md:19]: Please rephrase a bit.

File: safehttp/plugins/hsts/hsts.go [hunk 5]
```diff
@@ -28,7 +28,7 @@ import (
 type Plugin struct {
 	// MaxAge is the duration that the browser should remember
 	// that a site is only to be accessed using HTTPS. MaxAge
-	// must be positive.
+	// must be positive. Will be rounded to seconds before use.
```
Comment (mihalimara22) [inline safehttp/plugins/hsts/hsts.go:31]: "It will.."

File: docs/high-level-design.md (hunk 1 above)
Comment (mattiasgrenfeldt) [reply docs/high-level-design.md:77]: It will work once this is merged.

File: docs/plugins/hsts.md (hunk 4 above)
Comment (mattiasgrenfeldt) [reply docs/plugins/hsts.md:19]: I see that I missed a `be` in this sentence:
Must be enabled for this domain to **be** eligible to be added to the browser HSTS preload list.

Other than that, what should I rephrase? What is unclear?

File: docs/plugins/hsts.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+## Usage
+
+To construct the plugin with safe default settings, use: `hsts.NewPlugin()`.
+
+## Options
+
+**Option (Default value)**: Description.
+
+- **MaxAge (`2 years`)**: The amount of time that the browser should remember to only access this site using HTTPS. The default value is 2 years since this is part of the deployment recommendations of https://hstspreload.org/.
```
Comment (mihalimara22) [inline docs/plugins/hsts.md:17]: "...access a website using HTTPS.  The default value is 2 years as recommended in https://hstspreload.org/."

File: docs/plugins/hsts.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+- **IncludeSubDomains (`enabled`)**: When this is enabled, the browser knows that all subdomains should also only be accessed via HTTPS.
+- **Preload (`disabled`)**: Must be enabled for this domain to be eligible to be added to the browser HSTS preload list. For more info, see: https://hstspreload.org/. This is disabled by default so that your domain can't be added to the preload list unintentionally.
```
Comment (mihalimara22) [inline docs/plugins/hsts.md:19]: This sentence is a bit hard to follow. Maybe something like: "If enabled, a domain is eligible to be added as part of the browser HSTS preload list. This is disabled by default to prevent adding domains to the preload list unintentionally."

File: docs/plugins/hsts.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+- **BehindProxy (`disabled`)**: If this server is placed behind a proxy that terminates HTTPS then only HTTP traffic will be received. This option disables redirection of HTTP to HTTPS and always sends the `Strict-Transport-Security` header, even over HTTP.
```
Comment (mihalimara22) [inline docs/plugins/hsts.md:20]: The first sentence is a bit unclear to me, please rephrase a bit

File: docs/plugins/hsts.md (hunk 4 above)
Comment (mihalimara22) [reply docs/plugins/hsts.md]: I added some comments to each option, sorry it was unclear :)

File: docs/high-level-design.md (hunk 1 above)
Comment (kele) [reply docs/high-level-design.md:77]: Are relative links possible?

File: docs/plugins.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
+_Authors: grenfeldt@google.com_
+
+# Plugins
+
+The framework will support plugins that run before and after the request handlers. They provide a flexible way to address security issues by setting non-overridable security headers or interrupting and responding to incoming requests before they reach the handler.
```
Comment (kele) [inline docs/plugins.md:5]: Please wrap to 80 lines or so. Here and all other places.

Comment (kele) [inline docs/plugins.md:5]: It is okay to write documentation in a somewhat wishful way. i.e. do: "Framework supports plugins..." instead of "Framework *will* support plugins...". Unless this is a design doc, but in this case, this is a user-facing piece of documentation.

Also, make sure you introduce the details in the right places. Not all plugins will run before and after the request handlers, and this is not the level of abstraction that's appropriate. Interceptors (parts of plugins) will do that, so omit it for now.

File: docs/high-level-design.md (hunk 1 above)
Comment (mattiasgrenfeldt) [reply docs/high-level-design.md:77]: Yup, https://github.blog/2013-01-31-relative-links-in-markup-files/
Fixed!
